### DISCUSSION TITLE
What is the maximum size of a task?

### DISCUSSION THREAD

**Discussion by piavgh** [hibiken/asynq#460] [Q&A] [answered]
Hi everyone,

What is the maximum size of a task that I can put into the queue? Does it follow the Redis maximum key size?

Thanks

**Comment by hibiken** [comment, marked as answer]
@piavgh Thank you for opening a discussion!

It's probably closer to whatever value size limit for redis value is. (Each task's data, including its metadata are stored under a single redis key). That being said, we recommend you keep the memory footprint of each task small and store only the minimal data need to handle data (e.g. user ID instead of a marshaled user object).

LMK if you have more questions! Happy to help.
